modules/apps/ccv/types/expected_keepers.go

```diff
@@ -0,0 +1,17 @@
+package types
+
+// StakingKeeper defines the contract expected by parent-chain ccv module.
+// StakingKeeper is responsible for keeping track of latest validator set of all baby chains
+type StakingKeeper interface {
+	GetValidatorSetChanges(chainID string) []abci.ValidatorUpdate
+	// This method is not required by CCV module explicitly but necessary for init protocol
+	GetInitialValidatorSet(chainID string) []sdk.Tx
+}
+
+// CNSKeeper defines the contract expected by parent-chain ccv module
+// CNSKeeper is responsible for verifying that chain creator is authorized to create a chain with given chain-id
+type CNSKeeper interface {
```

Comment:
This will allow CNS to enforce that only chain-id owners create new chains with given chain-id. Other parent chains may choose to implement a no-op here, in which case CCV will simply ensure that a given baby chain-id is not already in use before accepting it.